core/src/main/java/org/apache/calcite/rel/core/AggregateCall.java

```diff
@@ -110,6 +111,7 @@ private AggregateCall(SqlAggFunction aggFunction, boolean distinct,
   public static AggregateCall create(SqlAggFunction aggFunction,
       boolean distinct, List<Integer> argList, int groupCount, RelNode input,
       RelDataType type, String name) {
+    distinct = aggFunction.getDistinctOptionality() != Optionality.IGNORED && distinct;
```

Comment:
Can we move this common logic into a method to make it more readable ?

Reply:
well,  Let me modify it